pkg/query-service/telemetry/telemetry.go

```diff
@@ -73,28 +74,41 @@ func (a *Telemetry) IsSampled() bool {
 
 }
 
-func (telemetry *Telemetry) CheckSigNozMetricsV2(compositeQuery *model.CompositeMetricQuery) bool {
-
-	signozMetricsNotFound := false
-
-	if compositeQuery.BuilderQueries != nil && len(compositeQuery.BuilderQueries) > 0 {
-		if !strings.Contains(compositeQuery.BuilderQueries["A"].MetricName, "signoz_") && len(compositeQuery.BuilderQueries["A"].MetricName) > 0 {
-			signozMetricsNotFound = true
+func (telemetry *Telemetry) CheckSigNozSignals(postData *v3.QueryRangeParamsV3) (bool, bool, bool) {
+	signozLogsUsed := false
+	signozTracesUsed := false
+	signozMetricsUsed := false
+
+	if postData.CompositeQuery.QueryType == v3.QueryTypeBuilder {
+		for _, query := range postData.CompositeQuery.BuilderQueries {
+			if query.DataSource == v3.DataSourceLogs {
+				signozLogsUsed = true
+			} else if query.DataSource == v3.DataSourceTraces {
+				signozTracesUsed = true
+			} else if query.DataSource == v3.DataSourceMetrics {
+				signozMetricsUsed = true
+			}
 		}
-	}
-
-	if compositeQuery.PromQueries != nil && len(compositeQuery.PromQueries) > 0 {
-		if !strings.Contains(compositeQuery.PromQueries["A"].Query, "signoz_") && len(compositeQuery.PromQueries["A"].Query) > 0 {
-			signozMetricsNotFound = true
+	} else if postData.CompositeQuery.QueryType == v3.QueryTypePromQL {
+		for _, query := range postData.CompositeQuery.PromQueries {
+			if strings.Contains(query.Query, "signoz_") && len(query.Query) > 0 {
```

Comment:
@makeavish this `signoz_` check is to identify APM metrics being used at serviceOverview pages. This check should be included for `QueryTypeBuilder` also.